dspace-api/src/main/java/org/dspace/external/provider/impl/SHERPAv2JournalDataProvider.java

```diff
@@ -0,0 +1,358 @@
+     * @return HttpGet object to be executed by the client
+     * @throws URISyntaxException
+     */
+    private HttpGet constructHttpGet(String query, int start, int limit) throws URISyntaxException {
+        // Build URL based on search query
+        URIBuilder uriBuilder = new URIBuilder(url);
+        uriBuilder.addParameter("item-type", "publication");
+        uriBuilder.addParameter("filter", "[[\"title\",\"contains word\",\"" + query + "\"]]");
+        uriBuilder.addParameter("format", "Json");
+        // Set optional start (offset) and limit parameters
+        if (start >= 0) {
+            uriBuilder.addParameter("offset", String.valueOf(start));
+        }
+        if (limit > 0) {
+            uriBuilder.addParameter("limit", String.valueOf(limit));
+        }
+        if (StringUtils.isNotBlank(apiKey)) {
+            uriBuilder.addParameter("api-key", apiKey);
+        }
+
+        // Create HTTP GET object
+        HttpGet method = new HttpGet(uriBuilder.build());
+
+        // Set connection parameters
+        int timeout = 5000;
+        method.setConfig(RequestConfig.custom()
+            .setConnectionRequestTimeout(timeout)
+            .setConnectTimeout(timeout)
+            .setSocketTimeout(timeout)
+            .build());
+        return new HttpGet(uriBuilder.build());
+    }
+
+    @Override
+    public boolean supports(String source) {
+        return StringUtils.equalsIgnoreCase(sourceIdentifier, source);
+    }
+
+    /**
+     * Get number of results returned from a SHERPA v2 publication search
+     * @param query The query to be search on and give the total amount of results
+     * @return int representing number of journal results
+     */
+    @Override
+    public int getNumberOfResults(String query) {
+        HttpGet get = null;
+        try {
+            get = constructHttpGet(query);
+            HttpClient hc = new DefaultHttpClient();
+            HttpResponse response = hc.execute(get);
+            if (response.getStatusLine().getStatusCode() == 200) {
+                // Read response and parse as SHERPAResponse object
+                InputStream content = null;
+                try {
+                    content = response.getEntity().getContent();
+                    SHERPAResponse sherpaResponse = new SHERPAResponse(content, SHERPAResponse.SHERPAFormat.JSON);
+                    if (CollectionUtils.isNotEmpty(sherpaResponse.getJournals())) {
+                        return sherpaResponse.getJournals().size();
+                    }
+                } catch (IOException e) {
+                    log.error("Error reading input stream for SHERPAResponse: " + e.getMessage());
+                } finally {
+                    if (content != null) {
+                        content.close();
+                    }
+                }
+            }
+        } catch (IOException | URISyntaxException e) {
+            log.error("SHERPA/RoMEO query failed: ", e);
```

Comment:
I don't think we can.
This class and its methods implement ExternalDataService interface (via extending ExternalDataServiceImpl). The abstract methods don't have any throw statements in its signatures. I would be hesitant to include a refactor of this interface and abstract impl in this PR as it would have consequences for all other providers.